sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_value.hpp

```diff
@@ -744,11 +821,24 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
       : public _detail::AmqpCollectionBase<std::vector<std::uint8_t>, AmqpBinaryData> {
   public:
     AmqpBinaryData() : AmqpCollectionBase(){};
+    virtual ~AmqpBinaryData() = default;
```

Comment:
Can we/should we add a virtual destructor to the AmqpCollectionBase instead?
If we don't, we are sure that no code holds a pointer to AmqpCollectionBase, and only to these derived classes which add virtual destructors, correct?

Reply:
BTW, if that is the case, maybe we can add protected non-virtual constructor to AmqpCollectionBase? That would ensure that no one holds that by a pointer to base class (maybe with the exception of smart pointers which can do magic and safely hold pointers to the base class without public destructor).

Reply:
Because AmqpCollectionBase's destructor is protected, it cannot be virtual. clang-tidy on Linux complains if I mark it as virtual (the reason that you have virtual dtors is to guard against torn destructors, but with a protected base class destructor, there is no possibility of a torn destructor).

Reply:
FWIW, I just searched for ~AmqpCollectionBase, and it found no instances

Reply:
Ah, I haven't noticed that this very PR adds it as `protected` 🤦